de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/inventory/impl/InventoryAllocationDestination.java

```diff
@@ -185,7 +189,7 @@ public IAllocationResult load(final IAllocationRequest request)
 
 			//
 			// Get/create the inventory line based on the info from material receipt and request
-			final I_M_InventoryLine inventoryLine = getCreateInventoryLine(receiptLine, hu.getM_HU_ID(), movementDate);
+			final I_M_InventoryLine inventoryLine = getCreateInventoryLine(receiptLine, topLevelHU.getM_HU_ID(), movementDate);
```

Comment:
Couldn't it happen that this method is called twice with two different `request`s whose respective hus have the same top level HU?
In this case we would still run into the UC, right?

Reply:
I tested this case and it's working correctly. If a child of the same toplevel HU comes, the qty of the xisting line is modified. No new line is created